cpp/lib/result_set.cpp

```diff
@@ -4,382 +4,283 @@
 
 #include "result_set.h"
 #include "ResultSet.hpp"
+#include "ResultSetArrow.hpp"
+#include "ResultSetJson.hpp"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-    void * rs_create_with_json_result(
+    result_set_ptr rs_create_with_json_result(
         cJSON * json_rowset,
         SF_COLUMN_DESC * metadata,
-        QueryResultFormat_t * query_result_format,
+        QueryResultFormat_t query_result_format,
         const char * tz_string
     )
     {
-        switch (*query_result_format)
+        switch (query_result_format)
         {
-            case ARROW_FORMAT:
-                return rs_arrow_create_with_json_result(json_rowset, metadata, tz_string);
-            case JSON_FORMAT:
-                return rs_json_create(json_rowset, metadata, tz_string);
+#ifndef SF_WIN32
```

Comment:
we don't support ARROW on windows?

Reply:
That's for Windows 32-bit only. Arrow doesn't support win 32-bit build when we adding support for arrow format. Haven't tried with latest version but the arrow issue still open so unlikely it's solved.
https://github.com/apache/arrow/issues/32111